pdns/recursordist/rec-xfr.hh

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include "config.h"
+
+#include <condition_variable>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "iputils.hh"
+#include "lock.hh"
+#include "logr.hh"
+#include "dnsrecords.hh"
+#include "rust/lib.rs.h"
+
+class DNSName;
+class SOARecordContent;
+struct FWCatz;
+
+// Please make sure that the struct below only contains value types since they are used as parameters in a thread ct
```

Comment:
```suggestion
// All members of this struct must be copyable, as they are used as parameters in a thread constructor
```
I think the important attribute here is that the members of the struct be *copyable*, not that they are *value* types (in fact `std::shared_ptr` is sometimes considered a reference type). As long as all the members are safe to copy, that's sufficient.